File: seastarvideo/src/main/java/com/freewind/seastarvideo/api/ApiCode.java
package com.freewind.seastarvideo.api;

/**
 * author superK
 * update_at 2024/2/1
 * description 错误码
 */
public interface ApiCode {
      /**
       * 成功
       */
      int SUCCESS = 0;
      /**
       * 未指定通用错误码
       */
      int ERROR = 100300;
      String ERROR_STR = "未指定通用错误码";
      /**
       * 数据库错误、异常
       */
      int ERROR_SQL = 100301;
      /**
       * 数据记录未找到
       */
      int ERROR_DATA_NOT_FOUND = 100302;
      /**
       * 数据记录已存在
       */
      int ERROR_DATA_EXIST = 100303;
      /**
       * 无权限
       */
      int ERROR_NO_PERMISSION = 100304;
      /**
       * 未登录
       */
      int ERROR_NOT_LOGIN = 100305;
      /**
       * token已过期
       */
      int ERROR_TOKEN_EXPIRE = 100306;
      /**
       * token无效
       */
      int ERROR_TOKEN_INVALID = 100307;
      String ERROR_TOKEN_INVALID_STR = "token无效";
      /**
       * 网络错误、异常
       */
      int ERROR_NET = 100308;
      /**
       * 请求超时
       */
      int ERROR_REQUEST_TIMEOUT = 100309;
      /**
       * 请求参数不合法
       */
      int ERROR_PARAM_ILLEGAL = 100310;
      String ERROR_PARAM_ILLEGAL_STR = "请求参数不合法";
}

File: seastarvideo/src/main/java/com/freewind/seastarvideo/http/ApiCode.kt
package com.freewind.seastarvideo.http

/**
 * author superK
 * update_at 2024/2/1
 * description 错误码
 */
object ApiCode {

    /**
     * 成功
     */
    const val SUCCESS = 0

    /**
     * 未指定通用错误码
     */
    const val ERROR = 100300
    const val ERROR_STR = "未指定通用错误码"

    /**
     * 未登录
     */
    const val ERROR_AUTH_FAILED = 10041
    const val ERROR_AUTH_FAILED_STR = "未登录"

    /**
     * token 失效
     */
    const val ERROR_TOKEN_INVALID = 10042
    const val ERROR_TOKEN_INVALID_STR = "token 失效"

    /**
     * token 已过期
     */
    const val ERROR_TOKEN_TOKEN_EXPIRED = 10043
    const val ERROR_TOKEN_TOKEN_EXPIRED_STR = "token"

    /**
     * 请求参数不合法
     */
    const val ERROR_PARAM_ILLEGAL = 100310
    const val ERROR_PARAM_ILLEGAL_STR = "请求参数不合法"

    /**
     * http返回结果为空
     */
    const val ERROR_HTTP_RESULT_NULL = 100310
    const val ERROR_HTTP_RESULT_NULL_STR = "http返回数据为空"
}
